src/basic/uid-classification.c

```diff
@@ -109,23 +110,52 @@ const UGIDAllocationRange *acquire_ugid_allocation_range(void) {
         return &default_ugid_allocation_range;
 }
 
-bool uid_is_system(uid_t uid) {
-        const UGIDAllocationRange *defs;
-        assert_se(defs = acquire_ugid_allocation_range());
+bool uid_in_range(uid_t uid, UGIDRangeFlags flags) {
```

Comment:
i am still not convinced such a flags based solution really makes things more readable.